tap_github/authenticator.py

```diff
@@ -55,6 +58,59 @@ def is_valid(self) -> bool:
         return True
 
 
+def generate_jwt_token(
+    github_app_id: str,
+    github_private_key: str,
+    expiration_time: int = 600,
+    algorithm: str = "RS256",
+) -> str:
+    actual_time = int(time.time())
+
+    payload = {
+        "iat": actual_time,
+        "exp": actual_time + expiration_time,
+        "iss": github_app_id,
+    }
+    token = jwt.encode(payload, github_private_key, algorithm=algorithm)
+
+    if isinstance(token, bytes):
+        token = token.decode("utf-8")
+
+    return token
+
+
+def generate_app_access_token(
+    github_app_id: str,
+    github_private_key: str,
+    github_installation_id: Optional[str] = None,
+) -> str:
+    jwt_token = generate_jwt_token(github_app_id, github_private_key)
+
+    headers = {"Authorization": f"Bearer {jwt_token}"}
+
+    if github_installation_id is None:
+        list_installations_resp = requests.get(
+            url="https://api.github.com/app/installations", headers=headers
+        )
+        list_installations_resp.raise_for_status()
+        list_installations = list_installations_resp.json()
+
+        if len(list_installations) == 0:
+            raise Exception(f"No installations found for app {github_app_id}.")
+
+        github_installation_id = random.choice(list_installations)["id"]
```

Comment:
@ericboucher this is the line SonarCloud doesn't like. `choice` is already imported, so SonarCloud might think the following is okay?

```suggestion
        github_installation_id = choice(list_installations)["id"]
```